src/qml/qmlsettingparameter.cpp

```diff
@@ -0,0 +1,74 @@
+#include "qml/qmlsettingparameter.h"
+
+#include <qnamespace.h>
+
+#include <QObject>
+
+#include "moc_qmlsettingparameter.cpp"
+#include "util/assert.h"
+
+namespace mixxx {
+namespace qml {
+
+QmlSettingGroup::QmlSettingGroup(QQuickItem* parent)
+        : QQuickItem(parent) {
+}
+QmlSettingParameter::QmlSettingParameter(QQuickItem* parent)
+        : QmlSettingGroup(parent) {
+}
+
+void QmlSettingParameter::componentComplete() {
+    QmlSettingGroup::componentComplete();
+    QList<QmlSettingGroup*> pathItems;
+    auto* pParent = parentItem();
+    while (pParent != nullptr) {
+        auto* pManager = qobject_cast<QmlSettingParameterManager*>(pParent);
+        if (pManager) {
+            pManager->registerSettingParamater(this, pathItems);
+            return;
+        }
+        auto* pGroup = qobject_cast<QmlSettingGroup*>(pParent);
+        if (pGroup) {
+            pathItems.prepend(pGroup);
+        }
+        pParent = pParent->parentItem();
+    }
+    DEBUG_ASSERT(!"Couldn't find manager!");
+}
+
+QmlSettingParameterManager::QmlSettingParameterManager(QQuickItem* parent)
+        : QQuickItem(parent),
+          m_model(new QSortFilterProxyModel(this)) {
```

Comment:
`m_model` is a plain type while ur initializing from a ptr. 
Did you mean to do this?
```suggestion
        : QQuickItem(parent),
          m_model(this) {
```
otherwise you'd have a model thats parented to another model which is parented to `this` afaict.